dali/operators/reader/loader/webdataset/tar_utils.h

```diff
@@ -103,6 +106,12 @@ class DLL_PUBLIC TarArchive {
    */
   bool EndOfFile() const;
 
+  /**
+   * @brief Frees the filestream and prepeares the archive for later usage
+   *
```

Comment:
```suggestion
   * @brief Frees the underlying files tream
```
I don't think there's any "later usage" possible other than overwriting the stream with move assignment (if that's even legal).

Reply:
If the TarArchive can take a filestream only in the constructor then maybe make Close private and call it in the destructor. This way you wouldn't have a dangling tar archive with no file stream open.

Reply:
I think that having `Close` is still nice in case you just want to, well, _close_ the file (without immediately opening another one).

Reply:
it's kinda a preparation for the file descriptor management. I can throw those changes out if you want